_lab/lab01.md

```diff
@@ -0,0 +1,196 @@
+These are important classifications (look up some of the examples in this lesson to learn more), but they are difficult for humans to process visually. 
+Fortunately, you can make a data structure to help you keep track of this.
+
+The practical skills to be developed include:
+
+* Defining classes in Python  
+* Defining methods in Python classes
+
+Note: In general, it is always important to work on labs and reading early so you can gain the proper context and utilize our office hours to seek assistance / ask clarifying questions during the week before the deadline, if needed\!
+
+It is a good idea to read up on some tools we’ll use in this lab before you get started, specifically Chapter 1.4.2.1 (String Formatting) and 1.4.6 \- 1.4.6.1 (Object Oriented Programming).
+
+The main idea for this lab is to write a program that will organize Vulnerabilities into a Vulnerability list. The program should have the ability to add / remove / search for vulnerabilities.
+
+# Instructions
+
+We recommend that you organize your lab work for this lab in its own directory, e.g., lab01. This way all files for a lab are located in a single folder. Also, this will be easy to import various files into your code using the import \* from technique shown in lecture.
+
+You will need to create three files (note that the file names are case-sensitive when we import them in Python\!):
+
+* Vulnerability.py \- a file containing a class definition for a Vulnerability object.  
+* VulnDB.py \- a file containing a class definition for a VulnDB object.  
+* testFile.py \- a file with the tests for the class methods.
+
+## Vulnerability.py class
+
+The Vulnerability.py file will contain the definition of a Vulnerability record.
+
+We will define the Vulnerability attributes as follows:
+
+* **CVE** \- str that represents the unique CVE identifier of the vulnerability. Your program should ensure this field will be stored as a Python alpha-numeric string, with the same format as is stored in MITRE (two dashes separating three sections of the identifier CVE-2025-31910).  
+* **CWE** \- str that represents the MITRE weakness classification of the vulnerability.
+* **year** \- int that represents the release year of a vulnerability.
+
+You will write a constructor that allows the user to construct a Vulnerability object by passing in values for all of the fields. Your constructor should set these attributes with the value None by default.
+
+* \_\_init\_\_(self, cve, cwe, year)
+
+In addition to your constructor, your class definition should also support “setter” methods that can update the state of the Vulnerability objects:
+
+Each Vulnerability object should be able to call a method to_string() that you will implement, which returns a str with all the vulnerability attributes EXACTLY as shown (i.e., the string should contain all attributes in the following EXACT format "CVE" (CWE) - YEAR):
```

Comment:
I'd change all method references to be formatted as a code (i.e., using backticks), e.g., `.to_string()`